### PR TITLE
Fixed comments from pr

### PR DESCRIPTION
**Pull request by Taure**
I have done the changes needed.

Tested it and it runs ok.

I have one test that fails:
profile_merge_keys

Then it get some error at release.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,4 @@ logs
 priv/templates/*.dtl.erl
 ebin
 .edts
+rebar.lock
```
Comment (kovyl2404) [inline .gitignore]: Remove this, please. Lockfile should be included into repository.

File: src/rebar_file_utils.erl [hunk 2]
```diff
@@ -142,7 +144,7 @@ mv(Source, Dest) ->
     end.
 
 win32_robocopy_ok({ok, _}) -> true;
-win32_robocopy_ok({error, {Rc, _}}) when Rc<9, Rc==16 -> true;
+win32_robocopy_ok({error, {Rc, _}}) when Rc<9; Rc=:=16 -> true;
```
Comment (kovyl2404) [inline src/rebar_file_utils.erl:147]: Hm... Strange. This is my fault of course.

File: .gitignore (hunk 1 above)
Comment (Taure) [inline .gitignore]: I agree, thought I removed that... :( I fix it.

File: src/rebar_file_utils.erl (hunk 2 above)
Comment (Taure) [inline src/rebar_file_utils.erl:147]: If I remember it right ; is a or and , is and when it comes to guards.

**Comment by Taure** [comment]
How can I remove a file from a commit? Looks like it didn't remove .gitignore

File: src/rebar_file_utils.erl (hunk 2 above)
Comment (kovyl2404) [inline src/rebar_file_utils.erl:147]: Yep, sure, you are right. I wonder how it worked for me.

Comment (Taure) [inline src/rebar_file_utils.erl:147]: You had code in sh_loop that resolved this that I removed now when we have this function.

Hmm.. no. xcopy was working in a different way than mv function. Now they are more similar then before.

**Comment by kovyl2404** [comment]
You could do it at least two ways.
1. Delete line from gitignore. Commit file with `--amend` option (`git commit --amend`). This will fix your previous commit. Then you should push your branch with `-f` flag (force). This will overwrite your previous one.
2. In your repository do `git reset HEAD~1`. You will get back all changes from your previous commit. After that you are free to do necessary stuff and commit it again. As in first, you will have to do force push into remote repository.

**Comment by Taure** [comment]
Thank you. now it is removed.

**Comment by kovyl2404** [comment]
So what the deal with tests? I'm not sure I got you. All tests now are green for me on windows and linux both.

**Comment by kovyl2404** [comment]
One more question. May I squash two your commits into one?

**Comment by Taure** [comment]
For me this test fails:
profile_merge_keys

And then it comes to  dev_mode_release where it stops with this:
Errors generating release
          dev_mode_release230298: File not found: "dev_mode_release230298.app"

yes you can squash them

**Comment by kovyl2404** [comment]
Hm. We need investigate it. Which windows do you use? 8.1? I ran tests on 7 and server 12. Also, what is your project path looks like for? I mean is it long or short? Maybe it contains spaces or, I don't know, some local characters?

**Comment by Taure** [comment]
Using Windows 8.1 and erlang 17.5

This is what the log says from ct this is for test  dev_mode_release:

[0;32m===> Verifying dependencies...
[0m
[0;32m===> Compiling dev_mode_release230298
[0m
[0;32m===> Starting relx build process ...
[0m
[0;32m===> Resolving OTP Applications from directories:
          d:/project/rebar3k/_build/test/logs/ct_run.nonode@nohost.2015-06-03_21.30.17/lib.rebar.logs/run.2015-06-03_21.30.18/log_private/apps_dir1_224692
          d:/project/rebar3k/_build/test/logs/ct_run.nonode@nohost.2015-06-03_21.30.17/lib.rebar.logs/run.2015-06-03_21.30.18/log_private/apps_dir1_224692/_build/default/lib
          d:/erl6.4/lib[0m

[0;35m===> Missing beam file elf_format <<"d:/erl6.4/lib/hipe-3.11.3/ebin/elf_format.beam">>
[0m
[0;32m===> Resolved dev_mode_release230298-1.0.0
[0m
[0;32m===> Dev mode enabled, release will be symlinked
[0m
[0;32m===> Including Erts from d:/erl6.4
[0m

**Comment by kovyl2404** [comment]
Okay, I've merged you works and pushed it into my PR branch. You could track it [here](https://github.com/rebar/rebar3/pull/489) again. Thanks for effort.

**Comment by Taure** [comment]
Thank you. Should we continue here to resolve this issue or start a new issue to solve it?

**Comment by kovyl2404** [comment]
I sure we need to start investigate it and then, based on investigation result, we will decide what to do. About 'could not generate release' issue - I think you have `erts` without `hipe` support.

```
vkovalev@dijkstra:~/workspace/rebar3$ erl
Erlang/OTP 17 [erts-6.4.1] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false]
```

This is example output on my linux machine. Have you noticed `[hipe]` badge? Do you have this one?

**Comment by Taure** [comment]
It can be so that I am missing hipe compiler.

D:\project\rebar3k [windows-ct-fix +1 ~1 -0 !]> erl
Eshell V6.4  (abort with ^G)
1> application:start(hipe).
ok
2>

**Comment by kovyl2404** [comment]
Ha-ha. Windows `erl` doesn't show all those parameters. I gonna try to run test on erts without hipe, but on linux.

**Comment by Taure** [comment]
I did download the binary file from erlang.org maybe it is something with that.

**Comment by kovyl2404** [comment]
BTW, something similar was discussed [here](https://github.com/rebar/rebar/issues/291) in `rebar2`. So I believe this is not pure Windows issue. I'd post new report to rebar3 repo. But most likely, you have broken erlang installation.

**Comment by Taure** [comment]
Mm.. and that is kind off bad that both Esl and Ericsson is delivering broken erlang installations.

**Comment by Taure** [comment]
Maybe should try install erlang and this branch on my windows 7 machine and see if it fix the issue. What versionen of erlang do you use?

**Comment by kovyl2404** [comment]
On Windows 7 I use erlang 17.5 (erts 6.4) from erlang solutions.

**Comment by Taure** [comment]
Will look at it tomorrow. Need to fix things before going to sleep.

**Comment by kovyl2404** [comment]
Post new issue to original `rebar3` project. I believe, guys could help you more quickly than me =)

**Comment by kovyl2404** [comment]
This one was merged manually.

**Comment by Taure** [comment]
Thank you for the help and feedback.

**Pull request closed without merging**